### ISSUE TITLE
pytorch==0.4  loss is nan

### ISSUE THREAD

**Issue by Dorozhko-Anton** [amdegroot/ssd.pytorch#240]
I applied modifications to use pytorch==0.4 from https://github.com/amdegroot/ssd.pytorch/issues/175

also I set  
``` export NCCL_P2P_DISABLE=1  ```
to unstuck K80 gpus

But the loss becomes nan after the first iteration.

Did someone encounter that problem?

```
iter 0 || Loss: 30.8207 ||           
timer: 1.1393 sec.          
iter 1 || Loss: nan ||
```

**Comment by pyaf** [comment]
try lower learning rate that what you were using then

**Comment by usherbob** [comment]
So, did you figure it out? @Dorozhko-Anton I met the same problem as you, and I used pytorch0.4.0.  I tried to lower the learning-rate, set it to 1e-4, 1e-5, but it did not work for me.

**Comment by 1243France** [comment]
i met this problem too. And I find i made a mistake when i am fixing the error in multiboxloss. you can review your code in multiboxloss.

**Comment by eng100200** [comment]
@1243France what was the issue in multiboxx loss?

**Comment by 1243France** [comment]
> 
> 
> @1243France what was the issue in multiboxx loss?

        loss_c = loss_c.view(num, -1)
        loss_c[pos] = 0
 line97,98  in ssd.pytorch/layers/modules/multibox_loss.py should make a swap in pytorch 0.4
 like above two line in mine.

**Comment by dazhangzhang** [comment]
@1243France   line97,98 in ssd.pytorch/layers/modules/multibox_loss.py should make a swap in pytorch 0.4
like above two line in mine.   
how do you make  a wap ?  can you make a example?  please

**Comment by imsjz** [comment]
> try lower learning rate that what you were using then

It worked for me when i set learning rate to 1e-5 training with VOC datasets.

**Comment by pengfeidip** [comment]
> > try lower learning rate that what you were using then
> 
> It worked for me when i set learning rate to 1e-5 training with VOC datasets.

But how many hours to train SSD with learning rate 1e-5? and what is your input-size?
